packages/components/src/date-time/time/time-input/index.tsx

```diff
@@ -108,6 +109,7 @@ export function TimeInput( {
 					className="components-datetime__time-field components-datetime__time-field-time" // Unused, for backwards compatibility.
 				>
 					<HoursInput
+						id={ id }
```

Comment:
No strong feelings here, but I guess the rationale for focusing the hours field was that it's the first field from a group of fields. Because from an accessibility standpoint, shouldn't every field have its own label? I'm not aware of a spec for timepickers, so if we do one one label with two fields, focusing the first one on click of the label makes sense to me. This is what folks do in React Aria too: https://react-spectrum.adobe.com/react-aria/TimeField.html

Reply:
I was only discussing the component prop design of arbitrarily passing through the `id` to the first input, which is separate from the label-click behavior.

That said, good pull with the React Aria one! I can get behind this implementation, with the multiple labeling, intentional ordering, and comma-separated (!) aria-labels. I'll try this on a component and see what the accessibility folks think.

Reply:
> I guess the rationale for focusing the hours field was that it's the first field from a group of fields.

That's the rationale, exactly.